### DISCUSSION TITLE
TableViewのCellへのデータの反映

### DISCUSSION THREAD

**Discussion by mahirooo** [lit-iphone-course/lit-iphone-course#19] [Question-original] [answered]
**現状の問題：**
- Todoに追加した最新のものひとつだけしかTableViewのCellに反映されない
- そもそも期限(date)が保存できてない⇨UIDatePickerの扱いがよくわかっておらずうまく保存できてない。

**解決したいこと：**
- 全てのTodoが表示されるようにしたい
- Cellの左側にTodoが、右側に期限が表示されるようにしたい

**やろうとしたこと：**
- Realmに保存された値を全て配列に入れて表示させようとした→エラーが出てうまくいかず。
- UIDatePickerで選択されたものをString型で保存しようとした→エラーが出てうまくいかず。

**調べた内容・記事**

- [https://naoya-ono.com/swift/realm-read/#2](url)
- [https://qiita.com/pe-ta/items/616e0dbd364179ca284b](url)
- [https://satoriku.com/hellorealm/](url)

Git⇨[https://github.com/mahirooo/OriginalTodo](url)

**Comment by txxxxc** [comment, marked as answer]
お疲れさまです！2点確認したいことがあります！
1. やろうとしたことに書いてあるエラーの内容を教えてもらえるとありがたいです!(エラーメッセージをコピペでOK)
2. 共有してもらったGitHubのリポジトリ(https://github.com/mahirooo/OriginalTodo)なんですけど、`Page not found`で見れないので確認してもらいたいです！(privateで作ってる場合は招待してもらえるとありがたいです！)

**Reply by mahirooo** [reply]
お疲れ様です！気付くの遅れてごめんなさい！！
こちら、「全てのTodoが反映されない」点は解消できました！

UIDatePickerで選択されたものをString型で保存しようとした→
→`Value of type 'UIDatePicker' has no member 'text'`
そもそもUIDatePickerで選択したものってInt...？String....？どっちで取り出せばいいんだろうってところでまず悩んでます。

publicで作ってるので見れるはず...！
https://github.com/mahirooo/OriginalTodo

**Reply by txxxxc** [reply]
お疲れさまです！
UIDatePickerで選択された日付はDate型なので、Date型をString型に変換してあげる必要があります！

変更方法は以下の記事を参考にやってみてくださいー！
https://tomato-develop.com/swiftui-date-string-cast/
